### PR TITLE
Remove support for EOL .NET 7.0

### PR DESCRIPTION
**Pull request by MichaelSimons**
Related to https://github.com/dotnet/dotnet-docker/issues/5336

### CODE REVIEW

**Comment by jander-msft** [comment]
Could you also delete https://github.com/dotnet/dotnet-docker/blob/nightly/eng/dockerfile-templates/monitor/Dockerfile.entrypoint.monitorV7

**Comment by MichaelSimons** [comment]
> Could you also delete https://github.com/dotnet/dotnet-docker/blob/nightly/eng/dockerfile-templates/monitor/Dockerfile.entrypoint.monitorV7

Done - thanks for pointing that out.

**Comment by lbussell** [comment]
Couple more references to 7.0 to remove: 
https://github.com/dotnet/dotnet-docker/blob/7b4b9a06391dda0b1880654516e4452dcbc22280/samples/build-for-a-platform.md#L86-L87
https://github.com/dotnet/dotnet-docker/blob/7b4b9a06391dda0b1880654516e4452dcbc22280/samples/complexapp/README.md#L186-L187
https://github.com/dotnet/dotnet-docker/blob/7b4b9a06391dda0b1880654516e4452dcbc22280/samples/kubernetes/dotnet-monitor/README.md#L29-L43
Several in this file, changing 7 -> 8 should work: https://github.com/dotnet/dotnet-docker/blob/7b4b9a06391dda0b1880654516e4452dcbc22280/samples/kubernetes/graceful-shutdown/graceful-shutdown.md#L21-L22
Update this condition to reference 6.0 instead of 7.0: https://github.com/dotnet/dotnet-docker/blob/7b4b9a06391dda0b1880654516e4452dcbc22280/tests/Microsoft.DotNet.Docker.Tests/MonitorImageTests.cs#L608-L617

**Review by mthalman — COMMENTED** [review]
Other references to clean up... do searches for:
* `== 7`
* `> 7`
* `>= 7`
* `<= 7`
* `!= 7`

**Comment by MichaelSimons** [comment]
@mthalman @lbussell - thanks for the feedback.  I believe I have address all of your comments.

File: samples/kubernetes/graceful-shutdown/graceful-shutdown.md [hunk 1]
```diff
@@ -122,8 +122,8 @@ The `IHostApplicationLifetime.ApplicationStopping` token will now be activated o
 
 ## Background services
 
-For background services ASP.NET provides [`IHostedService` interface and a `BackgroundService` base class](https://learn.microsoft.com/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-7.0) that allow for easy handling of `ApplicationStopping` event:
+For background services ASP.NET provides [`IHostedService` interface and a `BackgroundService` base class](https://learn.microsoft.com/aspnet/core/fundamentals/host/hosted-services) that allow for easy handling of `ApplicationStopping` event:
```
Comment (MichaelSimons) [inline samples/kubernetes/graceful-shutdown/graceful-shutdown.md:125]: I noticed you can make a versionless query and it defaults to the latest.  There are pros and cons to this but I think this is a better choice for this doc.

Comment (lbussell) [reply samples/kubernetes/graceful-shutdown/graceful-shutdown.md:125]: Agreed

**Review by mthalman — COMMENTED** [review]
Found some more things to get cleaned:

https://github.com/dotnet/dotnet-docker/blob/ea2167d44a8e0841c8ed09210dfa7a87dadae1bc/eng/dockerfile-templates/monitor/Dockerfile.envs#L5

https://github.com/dotnet/dotnet-docker/blob/ea2167d44a8e0841c8ed09210dfa7a87dadae1bc/eng/dockerfile-templates/monitor/Dockerfile.linux.extensions#L8

https://github.com/dotnet/dotnet-docker/blob/ea2167d44a8e0841c8ed09210dfa7a87dadae1bc/samples/kubernetes/dotnet-monitor/README.md?plain=1#L38

https://github.com/dotnet/dotnet-docker/blob/ea2167d44a8e0841c8ed09210dfa7a87dadae1bc/samples/kubernetes/resource-limits/README.md?plain=1#L48

https://github.com/dotnet/dotnet-docker/blob/ea2167d44a8e0841c8ed09210dfa7a87dadae1bc/tests/Microsoft.DotNet.Docker.Tests/SdkImageTests.cs#L76

**Comment by MichaelSimons** [comment]
> Found some more things to get cleaned:

I fixed these in https://github.com/dotnet/dotnet-docker/pull/5470/commits/43383ff93b2203ab7acecf01c259d527268d3db3 and https://github.com/dotnet/dotnet-docker/pull/5470/commits/ea2da8324b7c663ca7c8c850da2edbb6f451a50f along with a couple other references/issues I found.

File: samples/kubernetes/resource-limits/README.md [hunk 2]
```diff
@@ -45,13 +45,13 @@ View the sample app at http://localhost:8080/ or call `curl http://localhost:808
...
+{"runtimeVersion":".NET 8.0.5","osVersion":"Alpine Linux v3.19","osArchitecture":"X64","user":"root","processorCount":1,"totalAvailableMemoryBytes":78643200,"memoryLimit":104857600,"memoryUsage":34095104,"hostName":"dotnet-resource-limits-8685bd4b75-jltn5"}
 ```
 
 `processorCount` is display as `1`, as expected.
 
 Delete the resources (remote URL or local manifest).
 
 ```bash
-kubectrl delete -f https://raw.githubusercontent.com/dotnet/dotnet-docker/main/samples/kubernetes/resource-limits/resource-limits.yaml
+kubectl delete -f https://raw.githubusercontent.com/dotnet/dotnet-docker/main/samples/kubernetes/resource-limits/resource-limits.yaml
```
Comment (MichaelSimons) [inline samples/kubernetes/resource-limits/README.md:56]: This was an existing bug.   I found it while walking through the sample.  It was copied in the replicas-and-health sample.

**Review by mthalman — APPROVED** [review]

**Pull request merged**